### PR TITLE
Redo the visitor so that we can target children better

### PR DESCRIPTION
**Pull request by jimmyhchan**
This adds a buildStackItem pass to mimic what we were doing with Context and Instructions.
That pass should be temporary and should be replaced with some better APIs that only add the context that is necessary (TBD).

For examples:
- remove tdIndex from buildStackItem and do a pass that adds index number instead
- replace the namespace logic to something else
- replace escaplable Raw
- move the instruction logic into something else?

### CODE REVIEW

File: src/compiler/visitors/visitorKeys.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  let extraArgs = nativeSlice.call(arguments, 2);
+  if (node[1].tag_info.attributes) {
+    visit.apply(null, [['htmlAttributes'].concat(node[1].tag_info.attributes)].concat(extraArgs));
+  }
+  if (node[1].tag_contents) {
+    visit.apply(null, [['htmlTagContents'].concat(node[1].tag_contents)].concat(extraArgs));
+  }
+};
+
+let visitHtmlAttributeRHSs = function(node, visit) {
```
Comment (smfoote) [inline src/compiler/visitors/visitorKeys.js:35]: What's an RHS?

**Comment by smfoote** [comment]
Looks good to me. Do the tests pass? Has @prashn64 taken a look yet?

File: src/compiler/visitors/visitorKeys.js (hunk 1 above)
Comment (jimmyhchan) [inline src/compiler/visitors/visitorKeys.js:35]: right hand side

**Comment by jimmyhchan** [comment]
yes. tests's pass.   @prashn64  has not looked at it?

**Pull request merged**